src/AppBundle/Controller/SettingsController.php

```diff
@@ -93,22 +93,17 @@ public function profileEditAction(Request $request)
     {
         $user = $this->getUserWithData();
 
-        switch ($this->getUser()->getRoleName()) {
-            case EntityDir\User::ROLE_ADMIN:
-            case EntityDir\User::ROLE_AD:
-                $form = $this->createForm(FormDir\User\UserDetailsBasicType::class, $user, []);
-                $jmsPutGroups = ['user_details_basic'];
-                break;
-            case EntityDir\User::ROLE_LAY_DEPUTY:
-                $form = $this->createForm(FormDir\Settings\ProfileType::class, $user, ['validation_groups' => ['user_details_full']]);
-                $jmsPutGroups = ['user_details_full'];
-                break;
-            case EntityDir\User::ROLE_PA:
-            case EntityDir\User::ROLE_PA_ADMIN:
-            case EntityDir\User::ROLE_PA_TEAM_MEMBER:
-                $form = $this->createForm(FormDir\Settings\ProfileType::class, $user, ['validation_groups' => ['user_details_pa', 'profile_pa']]);
-                $jmsPutGroups = ['user_details_pa', 'profile_pa'];
-                break;
+        if ($this->isGranted(EntityDir\User::ROLE_ADMIN) || $this->isGranted(EntityDir\User::ROLE_AD)) {
+            $form = $this->createForm(FormDir\User\UserDetailsBasicType::class, $user, []);
+            $jmsPutGroups = ['user_details_basic'];
+        } else if ($this->isGranted(EntityDir\User::ROLE_LAY_DEPUTY)) {
+            $form = $this->createForm(FormDir\Settings\ProfileType::class, $user, ['validation_groups' => ['user_details_full']]);
+            $jmsPutGroups = ['user_details_full'];
+        } else if ($this->isGranted(EntityDir\User::ROLE_PA)) {
+            $form = $this->createForm(FormDir\Settings\ProfileType::class, $user, ['validation_groups' => ['user_details_pa', 'profile_pa']]);
+            $jmsPutGroups = ['user_details_pa', 'profile_pa'];
+        } else {
+            $this->createAccessDeniedException('User role not recognised');
```

Comment:
switch statement is much easier on the eye :)

Reply:
not that easy, I'm using "isGranted" that uses the inheritance now. a switch will require all the sub-roles to be defined and will be more difficult to maintain